assets/js/theme/common/product-card.js

```diff
@@ -0,0 +1,30 @@
+const focusedCardClass = 'focused-card';
+const focusableElementsSelector = '[href], button, input, textarea, select, details, [tabindex]';
+const previousCard = {
+    $previous: null,
+};
+
+const onCardFocus = (event) => {
+    const $element = $(event.target);
+
+    const $cardContainer = $element.parents('.card'); 
+
+    const isFocusInCard = $cardContainer.length;
+
+    const { $previous } = previousCard;
+    
+    if (isFocusInCard) {
+        $previous && $previous.removeClass(focusedCardClass); 
+        $cardContainer.addClass(focusedCardClass);
+        previousCard.$previous = $cardContainer;
+        return;
+    }
+
+    $previous && $previous.removeClass(focusedCardClass);
+};
+
+export default function () {
+    const $allFocusableElements = $(focusableElementsSelector); 
```

Comment:
Maybe it would be better idea to listen focus events only on card focusable elements?

Reply:
No, because we need to hide buttons when we focusing not inside the card